lib/sdk/storage.js

```diff
@@ -84,7 +89,48 @@ function extension(filename, mimeType) {
     return ext;
 }
 
-// There is at least one worker that in some cases depends
+/**
+ * Validate url
+ * Url can be either a https url or datauri, throws if invalid
+ * @param {String} input 
+ */
+function validateSupportedUrl(input) {
```

Comment:
Then, when you look where `validateSupportedUrl` is used (https://github.com/adobe/asset-compute-pipeline/pull/20/files#diff-d40b97039ef41d1a5be7246fac0e0e4b65db319176b01bde4ba954917b5442c3R53), instead of creating a function that validates and then parse the protocol, you can change the structure of the if/else in https://github.com/adobe/asset-compute-pipeline/pull/20/files#diff-d40b97039ef41d1a5be7246fac0e0e4b65db319176b01bde4ba954917b5442c3R53 to only use the URL validation check (but this means changing the way you nested too).

The reason to do this is that:
- `validateSupportedUrl` is used only once as far (not counting unit tests);
- it's an additional layer for a function used once;
- restructuring the code where it is used can make it shorter. For instance, 

I won't block the PR longer for that, but I'd like others to chime in with opinions here, @pheenomenon @jdelbick @badvision.

Reply:
Actually I also have similar question, cuz we seems to have validation in each lay of function, and I don't really know if this is a common sense in our team to validate in all places. Ex. right now `validateSupportedUrl` is used in `engine.prepare()` and `getAsset()`.
But having the if/else structure for `validateSupportedUrl` make sense, since checking the protocol only gives more detail error on which protocol is it not valid for but it is not really that necessary.

Reply:
@tmathern  Actually I also have question on validation when changing the code, cuz we seems to have validation in each layer of function, and I don't really know if this is a common sense in our team to validate in all places. Ex. right now `validateSupportedUrl` is used in `engine.prepare()` and `getAsset()`.
But having the if/else structure for `validateSupportedUrl` make sense, since checking the protocol only gives more detail error on which protocol is it not valid for but it is not really that necessary.

Reply:
This here is a library. We use it in a way that we know, after going through engine, that our URL is valid and would not need more validation.

But since it is a library, in theory another person could use it a different way, and enter the flow at another level (eg. transformer, or directly storage). In those cases, you would need to have validation at those levels. That is why we have validation at various levels, to cover different use cases and entry point. Because, in theory, anyone could use this library in different ways.